packages/editor/src/store/private-selectors.js

```diff
@@ -49,3 +49,26 @@ export const getInsertionPoint = createRegistrySelector(
 export function getListViewToggleRef( state ) {
 	return state.listViewToggleRef;
 }
+
+export const getPageContentBlocks = createRegistrySelector(
+	( select ) =>
+		( state, rootClientId = '' ) => {
+			const { getBlockOrder, getBlockName } = select( blockEditorStore );
+			const contentIds = [];
+			for ( const clientId of getBlockOrder( rootClientId ) ) {
+				const blockName = getBlockName( clientId );
+				if (
+					blockName === 'core/post-title' ||
+					blockName === 'core/post-featured-image' ||
+					blockName === 'core/post-content'
+				) {
+					contentIds.push( clientId );
+				} else if ( blockName !== 'core/query' ) {
+					contentIds.push(
+						...getPageContentBlocks( state, clientId )
+					);
+				}
+			}
+			return contentIds;
+		}
+);
```

Comment:
This looks like a `block-editor` store selector, which should be memoized; otherwise, `clientIds` will be a new array on each call. 

Sidenote: Recursively calling memoized could impact performance - #58356.

Reply:
I don't like putting this in `@wordpress/block-editor` because I feel `block-editor` shouldn't know about "post title", "post featured image", "post content", the concept of page content, etc.

Unfortunately we can't combine `createSelector` and `createRegistrySelector`.

I'll go back to a solution that doesn't involve a new selector 👍 0af14addd13093038b7ca9e9f18a56eb3ae0a63f

Reply:
Good point.

> Unfortunately we can't combine createSelector and createRegistrySelector.

I can't find PR right now, but I thought @ellatrix or @jsnajdr resolved that issue.

Reply:
Yes, they can be combined now, since #57888:
```js
createRegistrySelector( ( select ) => createSelector( ( state, rootClientId = '' ) => {
  ...
} );
```

Reply:
The `useSelect` solution should be okay in this case. A `mapSelect` returns an array of `clientId` strings, which will pass shallow equality. There are few places in the core where we use similar "hacks"